server/api/utils/singletons/k8s.py

```diff
@@ -604,14 +607,15 @@ def ensure_configmap(
     @raise_for_status_code
     def get_configmap(self, name: str, resource: str, namespace: str = ""):
         namespace = self.resolve_namespace(namespace)
-        label_name = mlrun_constants.MLRunInternalLabels.resource_name
-        full_name = f"{resource}-{name}"
+        label_resource_name = mlrun_constants.MLRunInternalLabels.resource_name
+        label_resource_type = mlrun_constants.MLRunInternalLabels.resource_type
         configmaps_with_label = self.v1api.list_namespaced_config_map(
-            namespace=namespace, label_selector=f"{label_name}={full_name}"
+            namespace=namespace,
+            label_selector=f"{label_resource_name}={name},{label_resource_type}={resource}",
         )
         if len(configmaps_with_label.items) > 1:
             raise mlrun.errors.MLRunInternalServerError(
-                f"Received more than one config map for label: {full_name}"
+                f"Received more than one config map for labels: {name}, {resource}"
```

Comment:
this will result in the following string with only the values, and it is not clear what are the labels
```
Received more than one config map for labels: my-function, serving-conf
```